docs/build-configuration/environment-configuration.md

```diff
@@ -0,0 +1,29 @@
+---
+layout: "docs"
+title: Environment Configuration
+class: documentation
+permalink: /build/environment-configuration/
+published: true
+---
+
+You have access to certain environment options while configuring your Probo builds. These options are set as top level keys, on the same level as `assets` or `steps` in your `probo.yaml` file.
+
+## Allow Environment Access While Building
+
+We assume environments have confidential data and can be in potentially insecure states while building. Because of this, our default settings make sure environments are not made public until all of the steps have completed and the build is finished.
```

Comment:
Partially true. I thought it was more to do with the fact that the site wouldn't work and might be brittle but this slight inaccuracy may not be an issue.